### DISCUSSION TITLE
Is there a way to specify the version for all bundles, features and the updatesite only once?

### DISCUSSION THREAD

**Discussion by ingomohr** [eclipse-tycho/tycho#656] [Q&A] [answered]
Imagine you have a feature project, an updatesite project, 10 plugins and 10 test fragments.

When you abstract features like "pomless", you still have all the MANIFEST files etc which are to be updated when you raise the version for a next release.
Though setting the version to all these files can be done via Tycho, you still need the very same version in all these files.

**Question**:
Is there a way so that I can specify the version at exactly one place and have tycho (and the maven build) deal with setting the version automatically inside the Maven build?

**Comment by laeubi** [comment, marked as answer]
**Short answer:** No, there is no such thing, but you probably could use tycho-update-versions or similar.

**Some thoughts**
> which are to be updated when you raise the version for a next release

This "release version = artifact version" is really bad when it comes to the consumer of your artifacts. From an OSGi perspective you better use [semantic versioning](https://semver.org) instead of unnecessarily raise versions or even produce duplicated artifacts with the same content.

> I can specify the version at exactly one place and have tycho (and the maven build)

This is even not possible with maven, you can use CI-Friendly versions to some extend for this but there are some restrictions. We are currently working on this area, see also:

- https://github.com/apache/maven/pull/675
- https://github.com/apache/maven/pull/674

Beside that you still should consider semver as a better alternative here also.

**Reply by ingomohr** [reply]
Thank your for all the information!

We do semver already - which is why we have to raise the version
- e.g. from 2.1.0(snapshot/qualifier)
  - to 2.2.0(s/q) - for a backwards-compatible feature-update - or
  - to 3.0.0(s/q) - for an API-breaking change).

If we wouldn't raise the version, the user would get only a timestamp-update w/ the new build.

We update the version via [tycho-versions:set-version](https://www.eclipse.org/tycho/sitedocs/tycho-release/tycho-versions-plugin/set-version-mojo.html).

It's just that I feel there are too many places (all MANIFEST.MFs and all poml.xmls) that need to be updated - even w/ pomless builds.
